### ISSUE TITLE
Support for Calculator (keycode 0x0192) key?

### ISSUE THREAD

**Issue by bland328** [tekezo/Karabiner-Elements#1766]
**tl;dr: Is there a way to remap my new keyboard's Calculator key (keycode `0x0192`)?**

My new mechanical keyboard USB keyboard _(IKBC Series 108, FWIW)_ appears to Karabiner-Elements as both a keyboard and a mouse. Perhaps this is common _(or even_ standard, _for all I know)_, but it is new to me.

Three "special" media keys (Mute, Vol Dn, Vol Up) are seen as mouse events, and work perfectly "right out of the box" under Mojave, without Karabiner's help.

And when I configure Karabiner to modify events from the IKBC mouse device, it does see events for those three keys (`mute=0xe2; volume_decrement=0xea; volume_increment=0xe9`) in Karabiner-EventViewer.

So far, so good.

However, I have one more "special" key, labeled "Calculator."  I've confirmed on another system that it also generates a "mouse keystroke" _(for lack of knowing the right term)_ like the other three "special" keys. The keycode for this Calculator key is `0x0192`.

Is there any way to get Karabiner to map this key?

_Edit: To be clear, this Calculator key does_ not _generate any events in Karabiner-EventViewer, which leaves me thinking I'm out of luck, but I'm hoping that knowing the keycode might get me somewhere, unless macOS is getting in the way and consuming the event for some very "Apple" reason._

**Comment by tekezo** [comment]
Unfortunately, Karabiner-Elements cannot handle keys which are not shown on Karabiner-EventViewer.
These keys are dropped by macOS before apps (including Karabiner-Elements) catch the event.

**Comment by bland328** [comment]
Thank you! Makes sense, @tekezo, and I ended up handling that key with USB Overdrive.

**Comment by dufferzafar** [comment]
@bland328 Have you tried remapping the calculator key with `hidutil`? https://rakhesh.com/mac/using-hidutil-to-map-macos-keyboard-keys/

**Comment by tekezo** [comment]
Update:
The application launcher keys has been supported since v14.2.0.
https://karabiner-elements.pqrs.org/docs/releasenotes/#karabiner-elements-1420

**Comment by dufferzafar** [comment]
@tekezo, I can't actually use Karabiner anymore because of being on a work laptop, where a lot of permissions are restricted.

This is what worked for me: 

```
/usr/bin/hidutil property  --set '{"UserKeyMapping":[
    {
      "HIDKeyboardModifierMappingSrc": 0xC00000192,
      "HIDKeyboardModifierMappingDst": 0x70000006F
    }
]}'
```

To get the keycodes, [use this page](https://www.freebsddiary.org/APC/usb_hid_usages.php). Searching for `calculator` shows the key `AL Calculator` key marked as `0x0192` but it is under section 12 (which is `C` in hex) so that is where the `MappingSrc` entry comes from.

Similarly, search for `F20` and it shows up as `0x06F` under section `7`, giving us the `MappingDst` entry.

So, this remaps the Calendar key on Ducky keyboards to act as F20. You can change it to whatever you want, and then use that in other applications like HammerSpoon etc.
